### PR TITLE
You can now specify what URLs you want to download in a file

### LINKED ISSUE #529: [feature-request] start and end flags for a file
Something like these
```shell
annie.exe -F list.txt -start 2 -end 20
annie.exe -F list.txt -start 2 -end 2
annie.exe -F list.txt -start 2
annie.exe -F list.txt -end 20
```
Downloads only a subset of URLs from a list.
The `-F` option didn't work with `-start` and `-end` so I'm guessing this wasn't implemented.

### PR DESCRIPTION
**Pull request by Stegosawr**
Closes #529 

Merged all commits into one 
@iawia002

### CODE REVIEW

**Comment by iawia002** [comment]
the test failed:

```
    --- FAIL: TestParsingFile/start_from_x (0.10s)
        utils_test.go:633: Got: 608 - want: 632
```

File: utils/utils.go [hunk 1]
```diff
@@ -154,9 +154,6 @@ func ParseInputFile(r io.Reader) []string {
 	for scanner.Scan() {
 		totalLines++
 		universalURL := strings.TrimSpace(scanner.Text())
-		if universalURL == "" {
```
Comment (iawia002) [inline utils/utils.go]: if you remove this, there might be empty strings in temp

Comment (Stegosawr) [reply utils/utils.go]: yes, I know. But now that you can choose what urls you want this should be a problem. Also if I think about a file filled with url there shouldn't be any empty lines anyway?

**Comment by Stegosawr** [comment]
I am so sorry... fixed it now. Thanks for pointing it out and all the feedback you gave me :>

**Pull request merged**